oauth2_provider/views/oidc.py

```diff
@@ -121,3 +141,166 @@ def _create_userinfo_response(self, request):
         for k, v in headers.items():
             response[k] = v
         return response
+
+
+def validate_logout_request(user, id_token_hint, client_id, post_logout_redirect_uri):
+    """
+    Validate and Process an OIDC RP-Initiated Logout Request.
+    `(prompt_logout, (post_logout_redirect_uri, application))` is returned.
+
+    `prompt_logout` indicates whether the logout has to be confirmed by the user. This happens if the
+    specifications force a confirmation, or it is enabled by `ALWAYS_PROMPT_LOGOUT`.
+    `post_logout_redirect_uri` is the validated URI where the User should be redirected to after the
+    logout. Can be None. None will redirect to "/" of this app. If it is set `application` will also
+    be set to the Application that is requesting the logout.
+
+    The `id_token_hint` will be validated if given. If both `client_id` and `id_token_hint` are given they
+    will be validated against each other.
+    """
+    validator = oauth2_settings.OAUTH2_VALIDATOR_CLASS()
+
+    id_token = None
+    if id_token_hint:
+        # Note: The standard states that expired tokens should still be accepted.
+        # This implementation only accepts tokens that are still valid.
+        id_token = validator._load_id_token(id_token_hint)
+
+        if not id_token:
+            raise InvalidIDTokenError()
+
+        if id_token.user == user:
+            # The logout prompt can only not be shown if an ID Token is
+            # provided that matches the current user.
```

Comment:
An automatic logout (i.e. no prompt) is only allowed if an ID Token is provided that matches the current user.